cardinal/util.go

```diff
@@ -48,3 +52,34 @@ func isFatalError(err error) bool {
 	}
 	return true
 }
+
+func GetMessage[In any, Out any](wCtx WorldContext) (*MessageType[In, Out], error) {
+	var msg MessageType[In, Out]
+	msgType := reflect.TypeOf(msg)
+	tempRes, ok := wCtx.getMessageByType(msgType)
+	if !ok {
+		return nil, eris.Errorf("Could not find %q, Message may not be registered.", msg.Name())
+	}
+	var _ types.Message = &msg
+	res, ok := tempRes.(*MessageType[In, Out])
+	if !ok {
+		return &msg, eris.New("wrong type")
+	}
+	return res, nil
+}
+
+func GetTransactionReceiptsForTick(wCtx WorldContext, tick uint64) ([]receipt.Receipt, error) {
+	ctx, ok := wCtx.(*worldContext)
+	if !ok {
+		return nil, eris.New("error in test type assertion.")
+	}
+	return ctx.world.GetTransactionReceiptsForTick(tick)
+}
+
+func GetStoreManagerFromContext(wCtx WorldContext) gamestate.Manager {
+	return wCtx.storeManager()
+}
+
+func HandleQuery(wCtx WorldContext, query query, a any) (any, error) {
```

Comment:
If you really need to use these methods for test, then the test should be made as an internal test with the same package name so it has access to private methods, not by rewrapping a method that should be private into a public method.

Reply:
Doing this can cause unintentional side effects. Circular dependencies for one. It's lower risk to wrap it, that's my reasoning. But I can just go with what you want here.

Reply:
The risk of 3rd party user misusing a method that is supposed to be private is a much more significant problem [](edited-with-Axolo)

Reply:
agree to disagree. I think it's better to be private but not absolutely critical. But this is ultimately your engine, so I will following your philosophy. [](edited-with-Axolo)

Reply:
Hey @smsunarto moving relevant tests into internal causes circular dependencies with testutils and NewFixture which is a nightmare to resolve. Also the tester doesn't automatically bring the new test into the fold, it looks like it expects a specialized package cardinal_test. or some package name for _test for this to work. 

How about this, we punt this problem for now, I give HandleQuery some dangerous name like InternalHandleQuery or TestHandleQuery or something of that nature that will stop any user in his tracks from using it. I'll make a ticket for it because untangling this is looking to be a new small problem but sufficiently large enough to be it's own small ticket. 

I think to solve this problem we will have to bring NewTestFixture down into cardinal. 

Up to you. Let me know how you want to proceed.

Reply:
sure [](edited-with-Axolo)